### PR TITLE
refactor optimize GEMM on CPU tutorial

### PR DESCRIPTION
**Pull request by adstraw**
Updated "Loop Permutation" comments which reference `packedB` tensor before it is created by the script.

Updated "Array Packing" comments to more accurately describe the optimization. 

Replaced generic variable names (x, y, z) with more explicit variable names.

### CODE REVIEW

File: tutorials/optimize/opt_gemm.py [hunk 1]
```diff
@@ -227,43 +228,49 @@
...
-# some transformation we can get continuous access pattern. We can reorder a [16][16] array to
-# a [16/4][16][4] array, so that the access pattern of B will be sequential when grabing
-# the corresponding value from the packed array.
-#
+# We can use array packing to address the access pattern for B. Observe the array access pattern of
+# B after flattening which is not sequential as we iterate over the K dimension. We can reorder B
+# with dimensions [K][N] so that it has dimensions [N/bn][K][bn] where bn is the blocking factor and
+# also the vector size for both B in the inner loop.  This reorder splits N into two dimensions ---
+# bigN (N/bn) and littleN (bn) --- and the new dimensions [N/bn][K][bn] match the indexing of B
+# from outer to inner loops (no, ko, ki, ni) resulting in a sequentail access pattern for B after
```
Comment (gromero) [inline tutorials/optimize/opt_gemm.py:248]: s/sequentail/sequential/ ?

File: tutorials/optimize/opt_gemm.py [hunk 2]
```diff
@@ -227,43 +228,49 @@
 ###################################################################################################
 # Array Packing
 # -------------
-# Another important trick is array packing. This trick is to reorder the storage dimension of the
-# array to convert the continuous access pattern on certain dimension to a sequential pattern after
+# Another important trick is array packing. This trick is to reorder the storage dimension of an
+# array to convert the continuous access pattern on its dimensions to a sequential pattern after
```
[the remark this answers is not in the document]
Comment (hogepodge) [reply tutorials/optimize/opt_gemm.py]: I like the shorter sentence, but do you mean non-continuous though? Apologies for the late reply.

[the remark this answers is not in the document]
Comment (hogepodge) [reply tutorials/optimize/opt_gemm.py]: lgtm

**Review by tmoreau89 — APPROVED** [review]
Thanks for addressing the changes, LGTM!

**Comment by tmoreau89** [comment]
Thanks everyone who helped review this PR! And @adstraw for the PR! It is now merged.

**Pull request merged**